src/core/localize.rkt

```diff
@@ -28,15 +28,58 @@
 (define (batch-localize-error exprs ctx)
   (define errss
     (if (null? exprs) empty (compute-local-errors exprs ctx)))
+  (define result
   (for/list ([expr (in-list exprs)] [errs (in-list errss)])
     (sort
      (sort
       (reap [sow]
         (for ([(expr err) (in-hash errs)])
-          (unless (andmap (curry = 1) err)
-            (sow (cons err expr)))))
+          (sow (cons err expr))))
       expr<? #:key cdr)
      > #:key (compose errors-score car))))
+     result) 
+
+
+;;Returns a list of expressions sorted by increasing local cost in the form (subexpression, cost)
+(define (batch-localize-cost exprs ctx)
+  ;;Create a list of lists of subexpressions
+  (define subexprss
+    (for/list ([expr (in-list exprs)])
+      (all-subexpressions expr (context-repr ctx))))
+
+  ;Define a function to the get the cost of a expr with a repr
+  (define expr->cost  (platform-cost-proc (*active-platform*))); 
+
+  ;Create a list of (expr, cost, repr)
+  (define old-expr-cost-pair
+    (apply append
+     (map (lambda (listOfPairs)
+        (map (lambda (subexpr)        
+          (define expr (car subexpr))    
+          (define repr (cdr subexpr))         
+          (define cost (expr->cost expr repr))
+          (list expr cost repr)) 
+            listOfPairs))
+          subexprss)        
+   ))
+
+  ;create list of simplified expression
+  (define simplified-exprs  (simplify-batch (make-egg-query (map car old-expr-cost-pair) (*simplify-rules*))))
+  ;old=(expr cost repr)
+  ;new=(expr,expr,expr...)
+
+  ;Take the old list and the simplified lists and for each item in each takes the old cost subtracts the new cost and then returns a pair of the old-expr and the diff between them.
+  (define result
+  (sort
+    (for/list ([old old-expr-cost-pair]
```

Comment:
Also iterate over subexprss here! You can compute the original cost here!

Reply:
I can't because it is a list of lists.